components/organisms/ProfileTabBody.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Button,
+  CalendarSVG,
+  CogSVG,
+  EthTransparentSVG,
+  HeartSVG,
+  Modal,
+  Skeleton,
+} from "@ensdomains/thorin";
+import { ProfileRecordItem } from "../02-molecules";
+import { DatabaseIcon } from "../01-atoms";
+import { formatDate, formatHexAddress } from "@/lib/utils/formats";
+import { CoinInfo, DomainData } from "@/lib/domain-page";
+import { useAccount, useEnsName } from "wagmi";
+import { useState } from "react";
+import { EditResolverModalContent } from "./EditResolverModalContent";
+import { useRouter } from "next/router";
+import { excludeKeys } from "@/pages/domains/[name]";
+
+export interface ProfileTabProps {
+  domainData: DomainData | null;
+}
+
+export const ProfileTabBody = ({ domainData }: ProfileTabProps) => {
+  const [editResolverModalOpen, setEditResolverModalOpen] = useState(false);
+  const { address } = useAccount();
+
+  const router = useRouter();
+  const { name } = router.query; // Dynamic route parameter
+
+  const { data: authedUserName } = useEnsName({
+    address: address,
+  });
+
+  const showEditButton: boolean =
+    authedUserName === domainData?.owner || address === domainData?.owner;
+
+  const millisencondsToSeconds = (millisecodNumber: number): number =>
+    millisecodNumber / 1000;
+
+  const resolver = domainData?.resolver;
+
+  const addresses = resolver?.addresses;
+  const expiryDate = domainData?.expiryDate;
+
+  let filteredRecords: Record<string, string> = {};
+
+  if (domainData && typeof domainData.resolver.texts === "object") {
+    filteredRecords = Object.entries(domainData.resolver.texts)
+      .filter(([key]) => !excludeKeys.includes(key))
+      .reduce((obj: Record<string, string>, [key, value]) => {
+        obj[key] = value as string; // Type assertion to string
```

Comment:
the [graphql schema](https://github.com/blockful-io/external-resolver/blob/main/packages/gateway/src/types/metadata.ts#L23) defines it as string:string mapping, why cast it?

![image](https://github.com/user-attachments/assets/07c57673-db10-44d1-89a8-bd2527b592b3)